benchmark/buffers/buffer-compare-offset.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+var common = require('../common.js');
+
+var bench = common.createBenchmark(main, {
+  method: ['offset', 'slice'],
+  size: [16, 512, 1024, 4096, 16386],
+  millions: [1]
+});
+
+function compareUsingSlice(b0, b1, size, iter) {
+  const len = size >> 2;
+  for (var i = 0; i < iter; i++)
+    Buffer.compare(b0.slice(1, len), b1.slice(1, len));
+}
+
+function compareUsingOffset(b0, b1, size, iter) {
+  const len = size >> 2;
```

Comment:
interestingly.. forcing the optimization actually *slows* `b0.compare` down by a significant amount.